### PR TITLE
Add azurerm virtual network

### PR DESCRIPTION
**Pull request by eliecharra**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #1037 
| ❓ Documentation  | https://github.com/cloudskiff/driftctl-docs/pull/124

### CODE REVIEW

File: pkg/resource/azurerm/azurerm_virtual_network_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import (
+	"testing"
+
+	"github.com/cloudskiff/driftctl/test"
+	"github.com/cloudskiff/driftctl/test/acceptance"
+)
+
+func TestAcc_Azure_VirtualNetwork(t *testing.T) {
+	acceptance.Run(t, acceptance.AccTestCase{
+		TerraformVersion: "0.14.4",
```
Comment (sundowndev) [inline pkg/resource/azurerm/azurerm_virtual_network_test.go:12]: wondering if there's a particular reason to use 0.14.4 here ?

**Review by sundowndev — APPROVED** [review]

**Review by sundowndev — CHANGES_REQUESTED** [review]
*approved by mistake

File: pkg/resource/azurerm/azurerm_virtual_network_test.go (hunk 1 above)
Comment (eliecharra) [reply pkg/resource/azurerm/azurerm_virtual_network_test.go:12]: No, I'm gonna change it nice catch thanks 👍🏻

File: pkg/remote/azurerm/repository/network.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	ListAll(options *armnetwork.VirtualNetworksListAllOptions) virtualNetworksListAllPager
+}
+
+type virtualNetworksClientImpl struct {
+	client *armnetwork.VirtualNetworksClient
+}
+
+func (c virtualNetworksClientImpl) ListAll(options *armnetwork.VirtualNetworksListAllOptions) virtualNetworksListAllPager {
+	return c.client.ListAll(options)
+}
```
[the remark this answers is not in the document]
Comment (eliecharra) [reply pkg/remote/azurerm/repository/network.go:31]: This is for testing purposes exactly, we need to create an interface over the azure client and a dummy implementation to be able to mock the interface. This is ugly but I didn't find another way to mock azure SDK ...

File: go.mod [hunk 3]
```diff
@@ -6,6 +6,7 @@ require (
 	cloud.google.com/go/asset v0.1.0
 	github.com/Azure/azure-sdk-for-go/sdk/azcore v0.19.0
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v0.11.0
+	github.com/Azure/azure-sdk-for-go/sdk/network/armnetwork v0.3.0 // indirect
```
[the remark this answers is not in the document]
Comment (eliecharra) [reply go.mod:9]: Very true 👍🏻

File: pkg/remote/azurerm/azurerm_virtual_network_enumerator.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+		factory:    factory,
+	}
+}
+
+func (e *AzurermVirtualNetworkEnumerator) SupportedType() resource.ResourceType {
+	return azurerm.AzureVirtualNetworkResourceType
+}
+
+func (e *AzurermVirtualNetworkEnumerator) Enumerate() ([]*resource.Resource, error) {
+	resources, err := e.repository.ListAllVirtualNetwork()
```
[the remark this answers is not in the document]
Comment (eliecharra) [reply pkg/remote/azurerm/azurerm_virtual_network_enumerator.go]: I fixed this typo everywhere

**Review by sundowndev — APPROVED** [review]
👍🏻

**Pull request merged**